tests/unit/result/utils.test.js

```
@@ -0,0 +1,135 @@
+    });
+
+    expect(result).to.deep.equal([columnC, columnA, columnB]);
+  });
+});
+
+describe('convertThriftValue', () => {
+  it('should return value if type descriptor is missing', () => {
+    const value = 'test';
+    const result = convertThriftValue(undefined, value);
+    expect(result).to.equal(value);
+  });
+
+  [TTypeId.DATE_TYPE, TTypeId.TIMESTAMP_TYPE].forEach((typeId) => {
+    it(`should handle ${TTypeId[typeId]}`, () => {
+      const value = new Date();
+      const result = convertThriftValue({ type: typeId }, value);
+      expect(result).to.equal(value);
+    });
+  });
+
+  [TTypeId.UNION_TYPE, TTypeId.USER_DEFINED_TYPE].forEach((typeId) => {
+    it(`should handle ${TTypeId[typeId]}`, () => {
+      const value = {
+        n: 3.14,
+        s: 'test',
+        toString: () => JSON.stringify(this, null, 2),
+      };
+      const result = convertThriftValue({ type: typeId }, value);
+      expect(result).to.equal(String(value));
+    });
+  });
+
+  [TTypeId.DECIMAL_TYPE].forEach((typeId) => {
+    it(`should handle ${TTypeId[typeId]}`, () => {
+      [
+        ['3.14', 3.14],
+        [123, 123],
+      ].forEach(([value, expected]) => {
+        const result = convertThriftValue({ type: typeId }, value);
+        expect(result).to.equal(expected);
+      });
+    });
+  });
+
+  [TTypeId.BIGINT_TYPE].forEach((typeId) => {
+    it(`should handle ${TTypeId[typeId]}`, () => {
+      [
+        [BigInt(123), 123],
+        [new Int64(123), 123],
+        [123, 123],
+      ].forEach(([value, expected]) => {
+        const result = convertThriftValue({ type: typeId }, value);
+        expect(result).to.equal(expected);
+      });
+    });
+  });
+
+  [TTypeId.STRUCT_TYPE, TTypeId.MAP_TYPE].forEach((typeId) => {
+    it(`should handle ${TTypeId[typeId]}`, () => {
+      [
+        [undefined, undefined],
+        [{ s: 'test' }, { s: 'test' }],
+        [JSON.stringify({ s: 'test' }), { s: 'test' }],
+        ['{malformed json}', {}],
```

is there some consequence of just ignoring the data? Have you thought of what happens if we the data is returned in a weird way (interval: struc<interval, map> or any other weird data type that we know is not easily converted to JSON? Do we just throw the data away?

---

Currently yes, we just silently ignore weird data, and that's definitely not okay. We definitey should improve it, but it will be a separate PR

---

please create an issue and work on this. Let me know if you need help.

---

maybe add a comment on the tests that should be changed as they are testing current but wrong behavior